rosidl_generator_c/src/string_functions.c

```diff
@@ -84,12 +82,15 @@ rosidl_generator_c__String__assignn(
   if (n == SIZE_MAX) {
     return false;
   }
-  char * data = (char *)malloc(n + 1);
+  char * data = realloc(str->data, n + 1);
   if (!data) {
+    free(str->data);
+    str->data = NULL;
+    str->size = 0;
+    str->capacity = 0;
```

Comment:
This change will destroy the string if the function fails. The previous behavior would not change the existing string which I think is better behavior. What is the reason to change this?